homeassistant/components/scene/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""
+homeassistant.components.scene
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows users to set and activate scenes.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/scene/
+"""
+import logging
+from collections import namedtuple
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, SERVICE_TURN_ON, CONF_PLATFORM)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = 'scene'
+DEPENDENCIES = ['group']
+STATE = 'scening'
+
+CONF_ENTITIES = "entities"
+
+SceneConfig = namedtuple('SceneConfig', ['name', 'states'])
+
+
+def activate(hass, entity_id=None):
+    """ Activate a scene. """
+    data = {}
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+def setup(hass, config):
+    """ Sets up scenes. """
+
+    logger = logging.getLogger(__name__)
+
+    for entry in config:
```

Comment:
That feature was undocumented so I feel comfortable not making it backwards
compatible.

Paulus

On Tue, Feb 23, 2016 at 11:47 PM, sander76 <notifications@github.com> wrote:

> In homeassistant/components/scene/__init__.py
> <https://github.com/balloob/home-assistant/pull/1374#discussion_r53903819>
> :
>
> > +def activate(hass, entity_id=None):
> > +    """ Activate a scene. """
> > +    data = {}
> > +
> > +    if entity_id:
> > +        data[ATTR_ENTITY_ID] = entity_id
> > +
> > +    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
> > +
> > +
> > +def setup(hass, config):
> > +    """ Sets up scenes. """
> > +
> > +    logger = logging.getLogger(__name__)
> > +
> > +    for entry in config:
>
> Considering the following configuration file:
>
> scene test:
>   - name: All on
>     entities:
>       input_boolean.notify_home: on
>
> scene:
>   - name: All off
>     entities:
>       input_boolean.notify_home: off
>
> doing config.get(DOMAIN,{}) will get me only scene and not scene test.
> So I thought looping through all config entries will make sure I get hold
> of all scene domains and make sure they get a platform attached.
>
> I did test it locally and it works. (I also ran it through a debugger and
> the config does get the changes even though it is to be immutable.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1374/files#r53903819>.
>



-- 
PaulusSchoutsen.nl
It's nice to be important but it's more important to be nice.